internal/controller/controllers/agent_controller.go

```diff
@@ -905,6 +915,50 @@ func (r *AgentReconciler) updateIfNeeded(ctx context.Context, log logrus.FieldLo
 	return nil
 }
 
+func (r *AgentReconciler) getIgnitionToken(ctx context.Context, log logrus.FieldLogger, agent *aiv1beta1.Agent) (string, error) {
```

Comment:
The only thing you need here is `agent.Spec.IgnitionEndpointTokenReference` so pass that instead of the agent

Reply:
.